File: src/api/endpoints/auth/request/CreateApiTokenRequest.ts
import { TokenScope } from "../../../../models/ApiToken";

/**
 * Request body for creating an api token.
 * */
export default interface CreateApiTokenRequest {
  /**
   * Human-readable name for the token
   * */
  name: string;

  /**
   * Optional expiration date for the token
   * */
  expirationDate?: Date;

  /**
   * List of scopes for the token
   * */
  scopes: TokenScope[];
}

File: src-dotnet/api/endpoints/auth/request/CreateApiTokenRequest.cs
namespace MedrunnerApiClient.Api.Endpoints.Auth.Request;

    /// <summary>
    /// Request body for creating an API token.
    /// </summary>
    public class CreateApiTokenRequest
    {
        /// <summary>
        /// Human-readable name for the token
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Optional expiration date for the token
        /// </summary>
        public DateTime? ExpirationDate { get; set; }
    }
    